lib/channel.js

```diff
@@ -136,14 +216,14 @@ class Channel {
     try {
       result = await proc.invoke(context, args);
     } catch (error) {
-      const timedout = error.message === 'Timeout reached';
-      const code = timedout ? 408 : 500;
-      this.error(code, { callId, error, pass: timedout });
+      const timeout = error.message === 'Timeout reached';
+      const code = timeout ? 408 : 500;
+      this.error(code, { callId, error, pass: timeout });
       return;
     } finally {
       proc.leave();
     }
-    if (typeof result === 'object' && result.constructor.name === 'Error') {
+    if (typeof result === 'object' && result instanceof Error) {
```

Comment:
`instanceof` will not work in all cases because Error classes in different v8 isolated contexts are not `===`

Reply:
Ok

Reply:
We should also add `result !== null` here as it's also type of `object`.